.github/workflows/tests.yaml

```diff
@@ -47,3 +49,16 @@ jobs:
       mode: debug
       enables: --enable-cxx-modules
       enable-ccache: false
+      crypto_provider: GnuTLS
+  build_with_cxx_modules_openssl:
+    name: "Test with C++20 modules enabled (OpenSSL)"
+    uses: ./.github/workflows/test.yaml
+    strategy:
+      fail-fast: false
+    with:
+      compiler: clang++
+      standard: 23
+      mode: debug
+      enables: --enable-cxx-modules
+      enable-ccache: false
+      crypto_provider: OpenSSL
```

Comment:
piggyback the tests with OpenSSL / GnuTLS might not be a great idea. because, we only enable the test step in https://github.com/scylladb/seastar/blob/613d8b39e0d6e8af4b4aff5cffb639a7016c8f7c/.github/workflows/test.yaml#L110-L112 if "cxx-modules" is *not* enabled. in other words, these two jobs do not run any of the unit tests. so i'd suggest adding a dedicated job in `tests.yaml` for building with the OpenSSL backend, and keep the existing job of `build_with_cxx_modules` intact. 

please allow me to provide more context here: because C++20 modules is currently an experimental feature, not all Seastar facilities are exposed as in the "seastar" C++20 module at this moment, and we only have a single "hello_cxx_module" test for testing the build with C++20 module. none of the unit tests is built with the "seastar" C++20 module at the time of writing.